sky/utils/schemas.py

```diff
@@ -540,6 +540,19 @@ def get_config_schema():
                 **_NETWORK_CONFIG_SCHEMA,
             }
         },
+        'azure': {
+            'type': 'object',
+            'required': [],
+            'additionalProperties': False,
+            'properties': {
+                'storage_account': {
+                    'type': 'string'
+                },
+                'resource_group': {
```

Comment:
@cblmemo Good point.. Perhaps, `storage_resource_group`, `container_resource_group`, or `data_resource_group`?

Reply:
`{storage,blob}_resources_group` looks good to me. Here is another possible option. Lets discuss more in the meeting later ;)

```yaml
azure:
  blob:
    storage_account: xxx
    resources_group: xxx
```

Reply:
From offline discussion, we determined that we only ask for what `stroage_account` user wants to specify in `~/.sky/config.yaml`. This way, we can read the `resource_group` associated with the `storage_account`. This way, there isn't any false impression for the users and we won't need additional config of `blob`.

Reply:
Update: I guess we need to decide first between the two options below:
1. If we want to **let the users to provide non-existent storage account and resource group to `config.yaml`**, and therefore, creating them through Skypilot without using the default names, then we need to force the users to provide **both storage account and resource group** when using `config.yaml`, not just one of those two. This is for simplicity. The reason why I do not consider the case of user providing non-existent resource group only or non-existent storage account only is because it gives much more complex interface for the users, which is why there’s so much confusion around this additional layer of storage account and resource group compared to `s3` or `gcs`.
**pros:** we can provide end-to-end experience of setting up the storage account and resource group on top of container creation through Skypilot.
**cons:** we cannot just have `storage_account` under `config.yaml`. We need to force the users to specify resource group as well if they choose to use `config.yaml` for Azure storage.

2. If we want to **let the users provide only already existing storage account and resource group to `config.yaml`**, we can let the users to provide `storage_account` only in the `config.yaml` as we can infer resource group from the given `storage_account` through an API call.
**pros:** simplicity of user only having to provide the `storage_account`.
**cons:** lack of end-to-end experience of creating resource group and storage account through Skypilot. Although, I’m not quite convinced when this may be necessary.

I am leaning towards with option 1. And if we take this route, we can name the config values to be `storage_account` and `resource_group_storage` without adding additional layer of `blob`?

What do you think? @cblmemo @romilbhardwaj @Michaelvll

Reply:
How about this: Still keep the current config schema, e.g. only have `storage_account` under `azure` section, and do the following:

* If the storage account does not exist, create it **under our default resources group**
* If the storage account exists, we infer its resources group and use the value.

I'm thinking maybe we could hide the fact that "storage account needs a resources group to work". It feels too complex to expose to users for me. Open to any opinions though ;)

Reply:
As discussed offline following is the design we chose to interact with `config.yaml` with Azure blob storage:

1. User wants to rely on Skypilot interface so they don’t need to care for resource group or storage account: We create default resource group and storage account for them, and create container under those.
2. User wants to use externally created container either it being public or private: they provide the endpoint of the container url as `source:`(This is how other object storages are handled as well) at task yaml
3. User wants to create a container under the project group(storage account, resource group) they already have been using: user provides storage account to `config.yaml` and we can infer the resource group from it.
4. User wants to create new resource group and storage account through Skypilot, and then create container under those groups: Not supported for now.

Reply:
The change has been made at [720fe38](https://github.com/skypilot-org/skypilot/pull/3032/commits/720fe38ccd3699ec7cee2e6e7f0966f775a430fd) and all the scenarios above has been successfully tested.